EIPS/eip-5450.md

```diff
@@ -58,48 +58,50 @@ In the second validation phase control-flow analysis is performed on the code.
 
 *Terminating instructions* refers to the instructions either:
 
-- ending function execution: `RETF`, or
-- ending whole EVM execution: `STOP`, `RETURN`, `REVERT`, `INVALID`.
+- ending function execution: `RETF`, `JUMPF`, or
+- ending whole EVM execution: `STOP`, `RETURN`, `RETURNCONTRACT`, `REVERT`, `INVALID`.
 
-For each reachable instruction in the code the operand stack height is recorded.
-The first instruction has the recorded stack height equal to the number of inputs to the function type matching the code (`type[code_section_index].inputs`).
+Instructions in the code are scanned in a single linear pass over the code. For each instruction the operand stack height bounds are recorded as `stack_height_min` and `stack_height_max`.
 
-The FIFO queue *worklist* is used for tracking "to be inspected" instructions. Initially, it contains the first instruction.
+The first instruction's recorded stack height bounds are initialized to be equal to the number of inputs to the function type matching the code (`stack_height_min = stack_height_max = type[code_section_index].inputs`).
 
-While *worklist* is not empty, dequeue an instruction and:
+For each instruction:
 
-1. Determine the effect the instruction has on the operand stack:
-   1. **Check** if the recorded stack height satisfies the instruction requirements. Specifically:
-      - for `CALLF` instruction the recorded stack height must be at least the number of inputs of the called function according to its type defined in the type section.
-      - for `RETF` instruction the recorded stack height must be exactly the number of outputs of the function matching the code.
-   2. Compute new stack height after the instruction execution.
-2. Determine the list of successor instructions that can follow the current instructions:
+1. **Check** if this instruction has recorded stack height bounds. If it does not, it means it was neither referenced by previous forward jump, nor is part of sequential instruction flow, and this code fails validation. 
```

Comment:
We should mention the ordering of blocks as in eof.md, somewhere above: "Code basic blocks must be ordered in a way that every block is reachable either by a forward jump or sequential flow of instructions. In other words, there is no basic block reachable only by a backward jump."

Reply:
I mentioned it, but decided to put it into as a sub-point.

It feels not as important for implementers of the algorithm, and mainly important for code generators, so I thought it fits as a note. wdyt?

Reply:
totally can be a note :+1: